packages/flutter_tools/test/general.shard/build_system/targets/web_test.dart

```diff
@@ -0,0 +1,194 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter_tools/src/base/file_system.dart';
+import 'package:flutter_tools/src/base/platform.dart';
+import 'package:flutter_tools/src/base/process_manager.dart';
+import 'package:flutter_tools/src/build_system/build_system.dart';
+import 'package:flutter_tools/src/build_system/targets/dart.dart';
+import 'package:flutter_tools/src/build_system/targets/web.dart';
+import 'package:mockito/mockito.dart';
+import 'package:process/process.dart';
+
+import '../../../src/common.dart';
+import '../../../src/mocks.dart';
+import '../../../src/testbed.dart';
+
+void main() {
+  Testbed testbed;
+  Environment environment;
+  MockPlatform mockPlatform;
+
+  setUp(() {
+    mockPlatform = MockPlatform();
+    when(mockPlatform.isWindows).thenReturn(false);
+    when(mockPlatform.isMacOS).thenReturn(true);
+    when(mockPlatform.isLinux).thenReturn(false);
+    testbed = Testbed(setup: () {
+      environment = Environment(
+        projectDir: fs.currentDirectory,
+        outputDir: fs.currentDirectory,
+        buildDir: fs.currentDirectory,
+        defines: <String, String>{
+          kTargetFile: fs.path.join('lib', 'main.dart'),
+        }
+      );
+      environment.buildDir.createSync(recursive: true);
+    }, overrides: <Type, Generator>{
+      Platform: () => mockPlatform,
+    });
+  });
+
+  test('WebEntrypointTarget generates an entrypoint with plugins and init platform', () => testbed.run(() async {
+    environment.defines[kHasWebPlugins] = 'true';
+    environment.defines[kInitializePlatform] = 'true';
+    await const WebEntrypointTarget().build(environment);
+
+    final String generated = environment.buildDir.childFile('main.dart').readAsStringSync();
+
+    // Plugins
+    expect(generated, contains("import 'generated_plugin_registrant.dart';"));
+    expect(generated, contains('registerPlugins(webPluginRegistry);'));
+
+    // Platform
+    expect(generated, contains('if (true) {'));
+
+    // Main
+    expect(generated, contains('entrypoint.main();'));
+
+    // Import.
+    if (platform.isWindows) {
```

Comment:
According to the `setUp` above this will always be false. Only `isMacOS` is ever `true`. Or am I missing something?

Reply:
Fixed